@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -659,7 +659,7 @@ export class VisibleTileSet {
             // Remove all tiles that are still being loaded, but are no longer visible. They have to
             // be reloaded when they become visible again. Hopefully, they are still in the browser
             // cache by then.
-            if (!tile.isVisible && tile.tileLoader !== undefined && !tile.tileLoader.isFinished) {
+            if (!tile.isVisible && !tile.allGeometryLoaded) {
```

Comment:
is there more references to tile.tileLoader? otherwise we could now make it private too

Reply:
If TileDataSource would add getters on dataProvider and decoder, it could be constructed in the Tile without needing a signature change

Reply:
That's precisely what I meant by a workaround, those getters shouldn't be needed. Anyway, let's see what can be done, but in a separate PR.